packages/common-scripts/src/deploy.ts

```diff
@@ -0,0 +1,275 @@
+          tx_hash: secp256k1MultiSigConfig.TX_HASH,
+          index: secp256k1MultiSigConfig.INDEX,
+        },
+        dep_type: secp256k1MultiSigConfig.DEP_TYPE,
+      }
+    );
+  });
+
+  return txSkeleton;
+}
+
+async function completeTx(
+  txSkeleton: TransactionSkeletonType,
+  fromAddress: string,
+  feeRate?: bigint,
+  config?: Config
+): Promise<TransactionSkeletonType> {
+  const inputCapacity = txSkeleton
+    .get("inputs")
+    .map((c) => BigInt(c.cell_output.capacity))
+    .reduce((a, b) => a + b, BigInt(0));
+  const outputCapacity = txSkeleton
+    .get("outputs")
+    .map((c) => BigInt(c.cell_output.capacity))
+    .reduce((a, b) => a + b, BigInt(0));
+  const needCapacity = outputCapacity - inputCapacity + BigInt(10) ** BigInt(8);
+  txSkeleton = await common.injectCapacity(
+    txSkeleton,
+    [fromAddress],
+    needCapacity,
+    undefined,
+    undefined,
+    {
+      config: config,
+      enableDeductCapacity: false,
+    }
+  );
+  feeRate = feeRate || BigInt(1000);
+  txSkeleton = await common.payFeeByFeeRate(
+    txSkeleton,
+    [fromAddress],
+    feeRate,
+    undefined,
+    {
+      config: config,
+    }
+  );
+  return txSkeleton;
+}
+
+function calculateCodeHashByBin(scriptBin: Uint8Array): string {
+  const bin = scriptBin.valueOf();
+  return new utils.CKBHasher()
+    .update(bin.buffer.slice(bin.byteOffset, bin.byteLength + bin.byteOffset))
+    .digestHex();
+}
+
+async function getDataHash(outPoint: OutPoint, rpc: RPC): Promise<string> {
+  const txHash = outPoint.tx_hash;
+  const index = parseInt(outPoint.index, 10);
+  const tx = await rpc.get_transaction(txHash);
+
+  if (!tx) throw new Error(`TxHash(${txHash}) is not found`);
+
+  const outputData = tx.transaction.outputs_data[index];
+  if (!outputData) throw new Error(`cannot find output data`);
+
+  return new utils.CKBHasher().update(new Reader(outputData)).digestHex();
+}
+
+interface DeployOptions {
+  cellProvider: CellProvider;
+  scriptBinary: Uint8Array;
+  outputScriptLock: Script;
+  feeRate?: bigint;
+  config?: Config;
+}
+
+/**
+ * Generate txSkeleton for writing binary data to CKB, usually for deploying contracts.
+ * This generator only supports `SECP256K1_BLAKE160` and `SECP256K1_BLAKE160_MULTISIG` currently.
+ *
+ * @param options
+ */
+export async function generateDeployWithDataTx(
+  options: DeployOptions
+): Promise<TransactionSkeletonType> {
+  let txSkeleton = TransactionSkeleton({ cellProvider: options.cellProvider });
+
+  const fromAddress = generateAddress(options.outputScriptLock, {
+    config: options.config,
+  });
+
+  const output: Cell = {
+    cell_output: {
+      capacity: "0x0",
+      lock: options.outputScriptLock,
+    },
+    data: bytesToHex(options.scriptBinary),
+  };
+
+  txSkeleton = updateOutputs(txSkeleton, output);
+  txSkeleton = await completeTx(
+    txSkeleton,
+    fromAddress,
+    options.feeRate,
+    options.config
+  );
+
+  return updateCellDeps(txSkeleton, options.config);
```

Comment:
`updateCellDeps` after `payFee` may cause wrong fee calculation since the tx size has changed.  the deploy generator implemented here can refer to the implementation of [commons.transfer](https://github.com/nervosnetwork/lumos/blob/519ddf213b6164442e0e1fcdcbbb7d4b96456bed/packages/common-scripts/README.md), it return a `txSkeleton` that unpaid for fee